server/service/grpc/service.go

```diff
@@ -170,17 +170,37 @@ func (s *Service) CreateTable(ctx context.Context, req *metaservicepb.CreateTabl
 		return nil
 	}
 
-	procedure, err := factory.CreateCreateTableProcedure(ctx, procedure.CreateTableRequest{
-		Cluster:     c,
-		SourceReq:   req,
-		OnSucceeded: onSucceeded,
-		OnFailed:    onFailed,
-	})
-	if err != nil {
-		log.Error("fail to create table", zap.Error(err))
-		return &metaservicepb.CreateTableResponse{Header: responseHeader(err, "create table")}, nil
+	var p procedure.Procedure
+	if req.PartitionInfo != nil && len(req.PartitionInfo.GetNames()) == 0 {
```

Comment:
I guess it is not good to do validation for `CreateTableRequest` in the service layer because the service layer have no idea about the specific logic about what is necessary to create table and another validation is still required in the procedure layer even if it is done by the service layer.

However, it is not good to let the `factory` to take over such validation because of the same reason: the `factory` should not have any knowledge about that.

So I guess the best way is to let the specific procedure to do such validation, e.g. `CreateTableProcedure`.

Reply:
The content of this conversation is the same as that above, resolve it.